fixtures/get-bindings-proxy/tests/index.test.ts

```diff
@@ -223,15 +231,62 @@ describe("getBindingsProxy", () => {
  */
 async function startWorkers(): Promise<UnstableDevWorker[]> {
 	const workersDirPath = path.join(__dirname, "..", "workers");
-	const workers = await readdir(workersDirPath);
-	return await Promise.all(
-		workers.map((workerName) => {
+	const workerNames = await readdir(workersDirPath);
+	const workers: UnstableDevWorker[] = [];
+	for (const workerName of workerNames) {
+		const worker = await runWithCIDelay(async () => {
 			const workerPath = path.join(workersDirPath, workerName);
-			return unstable_dev(path.join(workerPath, "index.ts"), {
+			const worker = await unstable_dev(path.join(workerPath, "index.ts"), {
 				config: path.join(workerPath, "wrangler.toml"),
 			});
-		})
-	);
+			return worker;
+		});
+		workers.push(worker);
+	}
+	if (runningInCI) {
+		await Promise.all(workers.map((worker) => waitUntilReady(worker)));
+	}
+	return workers;
+}
+
+/**
+ * We've seen flakiness in CI (github actions) when spinning up multiple processes, this function
+ * tries to help such case by adding delays before and after such operations.
+ *
+ * Note: the delays are added only if the tests are run in CI as we haven't seen any such flakiness locally
+ *
+ * @param fn a function implementing an operation that we want to wrap in delays
+ * @param delay the amount of each delay in milliseconds
+ * @returns a promise yielding to the original function's result
+ */
+async function runWithCIDelay<T>(
+	fn: () => T | Promise<T>,
+	delay: number = 500
+): Promise<T> {
+	if (!runningInCI) {
+		return fn();
+	}
+
+	await setTimeout(delay);
+	const result = await fn();
+	await setTimeout(delay);
+
+	return result;
+}
+
+async function waitUntilReady(worker: UnstableDevWorker): Promise<Response> {
+	let response: Response | undefined = undefined;
+
+	while (response === undefined) {
+		await setTimeout(500);
+
+		try {
+			const url = `http://${worker.address}:${worker.port}`;
+			response = await fetch(url);
+		} catch (e) {}
```

Comment:
I feel kind of bad that this could loop forever (until the test timesout) without ever providing an error message that this is where it got stuck. But I see that you copied this implementation from elsewhere in the repo.

Not blocking, but it would be cool if we could have a timeout that causes this loop to break out and report a useful error.

Reply:
I'm actually changing this (since my changes don't seem to really help that much with the CI flakiness 🥲)